### ISSUE TITLE
Extending newline-per-chained-call

### ISSUE THREAD

**Issue by Nokel81** [eslint/eslint#10024]
**What rule do you want to change?**
Only allow `ignoreChainWithDepth` for the first line but not for others.

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new option: `onlyIgnoreForFirstLine: boolean`

**Please provide some example code that this change will affect:**

```js
//Valid use with {onlyIgnoreForFirstLine: true}
obj.map(x => x.foo).sort()
    .reduce((acc, cur) => Math.min(acc, cur));

obj.map(x => x.foo)
    .sort()
    .reduce((acc, cur) => Math.min(acc, cur));

//Valid use with {onlyIgnoreForFirstLine: false}
obj.map(x => x.foo)
    .sort().reduce((acc, cur) => Math.min(acc, cur));

//InValid use with {onlyIgnoreForFirstLine: true}
obj.map(x => x.foo)
    .sort().reduce((acc, cur) => Math.min(acc, cur));

//All these examples are also configured by `ignoreChainWithDepth`
//That option sets the number allowed on the first line
```

**What does the rule currently do for this code?**
Applies to all lines of chained commands

**What will the rule do after it's changed?**
If true then only apply to the first line

**Comment by platinumazure** [comment]
Have you looked at the [`ignoreChainWithDepth` option](https://eslint.org/docs/rules/newline-per-chained-call)?

**Comment by Nokel81** [comment]
I have but it would be nice to force the fact that the first line must have a method call on it

**Comment by platinumazure** [comment]
Hi @Nokel81, could you please edit the initial issue comment to reflect our proposed rule change process, as described in the link above? Thanks!

**Comment by Nokel81** [comment]
Done

**Comment by platinumazure** [comment]
Thanks!

Just to make it easier for others to dig in: Could you please add a comment in your examples about how the rule would otherwise be configured (e.g., the `ignoreChainWithDepth` setting)? Thanks!

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
